packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -631,7 +641,7 @@ export class Pipeline extends PipelineBase {
     }
 
     const app = this.supportScope();
-    const supportStackId = `cross-region-stack-${pipelineAccount}:${actionRegion}`;
+    const supportStackId = `cross-region-stack-${this.reuseCrossRegionSupportStacks === false ? pipelineStack.stackName : pipelineAccount}:${actionRegion}`;
```

Comment:
Since you default the field, it's never `undefined`, so let's just do a simple comparison here:

```suggestion
    const supportStackId = `cross-region-stack-${this.reuseCrossRegionSupportStacks ? pipelineAccount : pipelineStack.stackName}:${actionRegion}`;
```